src/wallet/rpcwallet.cpp

```diff
@@ -2389,33 +2389,72 @@ UniValue lockunspent(const JSONRPCRequest& request)
 
     RPCTypeCheckArgument(request.params[1], UniValue::VARR);
 
-    UniValue outputs = request.params[1].get_array();
-    for (unsigned int idx = 0; idx < outputs.size(); idx++) {
-        const UniValue& output = outputs[idx];
-        if (!output.isObject())
+    const UniValue output_params = request.params[1];
+
+    // Create and validate the COutPoints first.
+
+    std::vector<COutPoint> outputs;
+    outputs.reserve(output_params.size());
+
+    for (unsigned int idx = 0; idx < output_params.size(); idx++) {
+         const UniValue& param = output_params[idx];
+
+        if (!param.isObject()) {
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected object");
-        const UniValue& o = output.get_obj();
+        }
+
+        const UniValue& o = param.get_obj();
 
         RPCTypeCheckObj(o,
```

Comment:
In contrast to my comment above, it makes sense to keep these. `RPCTypeCheckObj` provides slightly more information (both the type expected and type received)